### ISSUE TITLE
Do I have to add a hook to `runner.py` in mmcv in order to be able to have different learning rate for different layers at different iteration?

### ISSUE THREAD

**Issue by yxchng** [open-mmlab/mmdetection#692]
Right now, all the layers seems to have the same learning rate at each iteration. I want to be able to have different learning rate for different layers.

From my research, I think I have to modify `mmcv` to so. Is there any other way to achieve what I want without touching `mmcv`?

On a side note, it seems like `mmcv` and `mmdetection` are quite tightly coupled. Why don't you just merge them? Tight coupling suggests that they should not be 2 different libraries. Otherwise, you should find a way to decouple them.

**Comment by yxchng** [comment]
Seems like I could do this without touching `mmcv` but I have to change the `build_optimizer(model, optimizer_cfg)` function like this:

```
base_lr = optimizer_cfg['lr']
base_wd = optimizer_cfg.get('weight_decay', None)
params = [{'params': model.backbone.parameters(), 'lr': base_lr*0.1, 'weight_decay': base_wd}, 
                 {'params': model.neck.parameters(), 'lr': base_lr*0.1, 'weight_decay': base_wd},
                 {'params': model.rpn_head.parameters(), 'lr': base_lr*0.1, 'weight_decay': base_wd},
                 {'params': model.bbox_head.parameters(), 'lr': base_lr*0.1, 'weight_decay': base_wd},
                 {'params': model.mask_head.parameters(), 'lr': base_lr, 'weight_decay': base_wd}]
optimizer_cls = getattr(torch.optim, optimizer_cfg.pop('type'))
return optimizer_cls(params, **optimizer_cfg)
```

Is this the way to do so? Is there any other way to pass `params` to the optimizer like in `torch`?

**Comment by hellock** [comment]
Yes this is the expected way to get it.

**Comment by rubeea** [comment]
Can anyone suggest on how can we find the optimal learning rate via using python packages such as:
https://github.com/davidtvs/pytorch-lr-finder
Where should I implement the functionality in the framework? Do I need to create a separate lr hook for this?

Any help will be greatly appreciated
Thanks
